okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -195,22 +195,26 @@ public HttpLoggingInterceptor setLevel(Level level) {
 
       String endMessage = "<-- END HTTP";
       if (logBody) {
-        BufferedSource source = responseBody.source();
-        source.request(Long.MAX_VALUE); // Buffer the entire body.
-        Buffer buffer = source.buffer();
-
-        Charset charset = UTF8;
-        MediaType contentType = responseBody.contentType();
-        if (contentType != null) {
-          charset = contentType.charset(UTF8);
+        if (response.code() == 204 || response.code() == 205) {
```

Comment:
Without this you get EOF because of transparent Gzip being unable to consume its header.